File: src/main/java/com/sobekcore/workflow/process/step/condition/Condition.java
package com.sobekcore.workflow.process.step.condition;

import com.sobekcore.workflow.process.step.condition.metadata.ConditionMetadata;
import com.sobekcore.workflow.process.step.condition.metadata.Metadata;
import com.sobekcore.workflow.utils.AnnotationLoader;

public record Condition(
    ConditionType type,

    ConditionData data
) {
    public static ConditionMetadata getMetadata(ConditionType type) {
        return AnnotationLoader.load(Condition.class.getPackageName(), Metadata.class, type);
    }
}

File: src/main/client/src/interfaces/process-step/condition.ts
import { ConditionType } from '@/enums/process-step/condition.ts';

export interface ConditionOption {
  label: string;
  value: string;
}

export interface ConditionData {
  link?: string;
  options?: ConditionOption[];
}

export interface Condition {
  type: ConditionType;
  data?: ConditionData;
}
